### PR TITLE
Update deprecated call of general_plain_validator_function (#676)

### PR DESCRIPTION
**Pull request by dslemusp**
https://docs.pydantic.dev/latest/api/pydantic_core_schema/#pydantic_core.core_schema.with_info_plain_validator_function

### CODE REVIEW

**Comment by hasB4K** [comment]
I would love this to merged! Most of my pytests warning on a project of mine are actually caused by this.

@roman-right, FYI, the deprecation was done by this PR: https://github.com/pydantic/pydantic-core/pull/980
And I think this PR can be merged like this since others projects just did exactly what @dslemusp is doing in this PR (a simple function replacement, see: https://github.com/vitalik/django-ninja/pull/922)

**Comment by roman-right** [comment]
Hi @hasB4K ,

Thank you for following up. I'll pick it up at the beginning of the next week.

File: beanie/odm/fields.py [hunk 1]
```diff
@@ -414,7 +414,7 @@ def __get_pydantic_core_schema__(
                     lambda instance: cls.serialize(instance)  # type: ignore
                 ),
             )
-            return core_schema.general_plain_validator_function(
+            return core_schema.with_info_plain_validator_function(
                 cls.build_validation(handler, source_type)
             )
```
Comment (hasB4K) [inline beanie/odm/fields.py:419]: I was reading the original code, and my IDE seems to complain that this code is unreachable.

Comment (dslemusp) [reply beanie/odm/fields.py:419]: Yeah that was also my case but as my PR is only related to deprecation I only changed the deprecated calls

**Comment by roman-right** [comment]
It looks good. I plan to have it merged at this weekend

**Review by rmariano — APPROVED** [review]

**Comment by roman-right** [comment]
Merged. It will be published today

**Pull request merged**